File: backjoon/greedy/Python/보물.py
def solve(N, A, B):
   
    result = 0

    for i in range(N) :
        result += (A[i] * B[i])


    print(result)

if __name__ == "__main__":
    
    N = int(input())
    A = sorted(list(map(int, input().split())))
    B = sorted(list(map(int, input().split())), reverse=True)

    solve(N, A, B)  # N과

File: backjoon/greedy/JS/보물.js
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "./dev/stdin" : "./input.txt")
  .toString()
  .trim()
  .split("\n");

//n이 5일거라 가정
const n = +input.shift();

//첫 번째 배열은 오름차순
let A = input[0]
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b);

//두 번째 배열은 내림차순
let B = input[1]
  .split(" ")
  .map(Number)
  .sort((a, b) => b - a);

let result = 0;

A.forEach((val, idx) => {
  result += val * B[idx];
});

console.log(result);
